Source/Fluxor/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -42,10 +42,59 @@ public static class ServiceCollectionExtensions
 				assembliesToScan: options.AssembliesToScan,
 				typesToScan: options.TypesToScan,
 				scanIncludeList: scanIncludeList);
-			services.AddScoped(typeof(IState<>), typeof(State<>));
+			services.AddRegistration(typeof(IState<>), typeof(State<>), options);
 			services.AddTransient(typeof(IStateSelection<,>), typeof(StateSelection<,>));
 
 			return services;
 		}
+
+		internal static IServiceCollection AddRegistration(this IServiceCollection services, Type serviceType, FluxorOptions options)
+		{
+			return options.RegistrationLifecycle switch {
+				LifecycleEnum.Scoped => services.AddScoped(serviceType),
+				LifecycleEnum.Singleton => services.AddSingleton(serviceType),
+				_ => services
+			};
+		}
```

Comment:
I think these would be better in `Fluxor\Extensions\IServiceCollectionExtensions.cs` as

`public static IServiceCollection Add(this IServiceCollection services, ......................., FluxorServiceLifetime lifetime)`

This would remove the code duplication in the ReduxDevTools project.

Reply:
I was going to make this implementation choice in the first place, but I second-guessed that you would have rather wanted more internal methods based upon the implementation style that I saw in the code.  (sigh)  no problem - easy enough to tweak and I like it better that way as well.

Reply:
consolidated into a single extension class and renamed